crates/qt-build-utils/src/builder/qmluri.rs

```diff
@@ -0,0 +1,34 @@
+// SPDX-FileCopyrightText: 2025 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+/// A builder for representing a QML uri
+#[derive(Clone)]
+pub struct QmlUriBuilder {
+    uri: Vec<String>,
+}
+
+impl QmlUriBuilder {
+    /// Construct a [QmlUriBuilder] from a given string
+    pub fn new(uri: impl IntoIterator<Item = impl Into<String>>) -> Self {
```

Comment:
We should probably have a `From<&str>` impl that splits on dots, so it's possible to do: `"com.kdab.com.cxx_qt".into()` when a QmlUri is needed.